### PR TITLE
python configuration prototype

### PR DESCRIPTION
**Pull request by codeboten**
Here's the start of the python prototype. Specifically here, I wanted to test out the ergonomics of using the dictionary object parsed from the config file. I've updated a couple of things in the schema/config file as a result which I'm not opposed to changing back but here's the list of changes:

1. changed propagators to an array of strings instead of objects. propagators are currently identified by a string in the spec. This means in theory users would be limited in what they can configure in a propagator, but that's the state of the world today.
2. renamed *_provider to the signal name instead. i think this makes understand the configuration a bit more ergonomic since for example we're configuring exporters for the signal, and then applying them to a specific processor. the user may not need to know this is configuring the provider specifically.
3. changed processors to the same format as the exporter with `type/identifier` naming

The code as is supports configuring a console exporter, and turning on/off the SDK. I will continue adding more interpretation of the configuration in the future. The only interface users would interact with is shown in `prototype.py` as the following:

```python
    otel.configure(otel.parse_and_validate_from_config_file(sys.argv[1], sys.argv[2]))
```

### CODE REVIEW

File: prototypes/python/otel.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+
+import importlib
```
Comment (ocelotl) [inline prototypes/python/otel.py:2]: Nit, use `from x import y`

**Comment by martinkuba** [comment]
Question about the processor names - what is the reason that the name includes the exporter name after the slash? I can see it as a more descriptive name, but the type of the processor will have to be parsed from the name.

**Comment by codeboten** [comment]
> Question about the processor names - what is the reason that the name includes the exporter name after the slash? I can see it as a more descriptive name, but the type of the processor will have to be parsed from the name.

It's probably not necessary, since the name is really identifying the type of processor this is and not used as a unique identifier anywhere else. I suppose it *could* make returning an error when configuring a specific processor a bit more easy to identify, but it's probably not needed at this time

File: prototypes/python/otel.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
+
+import logging
+from typing import List, Sequence, Tuple
+from pkg_resources import iter_entry_points
```
Comment (ocelotl) [inline prototypes/python/otel.py:4]: `pkg_resources` has been deprecated (see [this](https://github.com/open-telemetry/opentelemetry-python/pull/3047), and also [this](https://github.com/open-telemetry/opentelemetry-python/pull/3168)).

File: prototypes/python/otel.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    return entry_point.load()
+
+class Config:
+    def __init__(self, config=None) -> None:
+        self._config = config
+    
+    def _resource(self):
+        # resource detection
+        # attributes
+        from opentelemetry.sdk.resources import Resource
```
Comment (ocelotl) [inline prototypes/python/otel.py:40]: Why import here?

**Review by ocelotl — COMMENTED** [review]
Just some comments

File: prototypes/python/otel.py (hunk 2 above)
Comment (codeboten) [reply prototypes/python/otel.py:4]: Ah good to know. It's probably ok for the prototype but wouldn't want to ship this code :D

File: prototypes/python/otel.py (hunk 3 above)
Comment (codeboten) [reply prototypes/python/otel.py:40]: not necessary to have the import in the method, that was just me being lazy

**Comment by codeboten** [comment]
Closing in favour of https://github.com/open-telemetry/opentelemetry-configuration/pull/54

**Pull request closed without merging**